### PR TITLE
fix simple join parser error

### PR DESCRIPTION
**Pull request by pj-infest**

### CODE REVIEW

File: lib/sql_parser.js [hunk 1]
```diff
@@ -72,6 +76,13 @@ function getFrom(query) {
   return { db: path[1], collection: path[2], name: name[1] };
 }
 
+function getJion(query) {
```
Comment (uapasha) [inline lib/sql_parser.js]: little typo here

File: lib/sql_parser.js [hunk 2]
```diff
@@ -72,6 +76,13 @@ function getFrom(query) {
...
+  const from = query.match(/\sjoin\s+([\w([\])"'`.,/=<>\s]+$)/i)[1].replace(cutOff, '');
+  const name = from.match(/\s+as\s+(\w+)/i);
+  const path = from.slice(0, name.index).match(/(\w+)\/(\w+)/i);
```
Comment (uapasha) [inline lib/sql_parser.js]: `path`, `name` and `from` are very similar to `getFrom` related variables. Can you move them into separate function or constants to follow DRY

**Comment by oleg-rdk** [comment]
@striletskyy I would **very** recommend covering `parsing SQL query` functionality with a suite of unit tests.
This will give us:
 - assurance of no regression bugs in future.
 - acquaintance of existing parsing functionality.
 - huge time savings on manual testing (if not any other functionality is broken in any new MR). refers to p.1

**Comment by pj-infest** [comment]
@uapasha 
can not put all those functionality because there are problems with regexp.
Can not put part of regexp as param, and can not put as param whole regexp.

File: lib/sql_parser.js [hunk 3]
```diff
@@ -70,19 +70,22 @@ function getSelect(query) {
...
-  const path = from.slice(0, name.index).match(/(\w+)\/(\w+)/i);
+function getJoin(query) {
+  const from = query.match(/\sjoin\s+([\w([\])\"'`.,/=<>\s]+$)/i)[1].replace(cutOff, '');
+  const name = getParamName(from);
+  const path = parseQueryPartial(from, name);
   return path[2];
 }
 
+const getParamName = (queryPartial) => queryPartial.match(/\s+as\s+(\w+)/i);
+const parseQueryPartial = (partial, paramName) => partial.slice(0, paramName.index).match(/(\w+)\/(\w+)/i);
```
Comment (uapasha) [inline lib/sql_parser.js]: line is longer then 100 symbols

File: lib/sql_parser.js [hunk 4]
```diff
@@ -70,19 +70,22 @@ function getSelect(query) {
 }
 
 function getFrom(query) {
-  const from = query.match(/\sfrom\s+([\w([\])"'`.,/=<>\s]+$)/i)[1].replace(cutOff, '');
-  const name = from.match(/\s+as\s+(\w+)/i);
-  const path = from.slice(0, name.index).match(/(\w+)\/(\w+)/i);
+  const from = query.match(/\sfrom\s+([\w([\])\"'`.,/=<>\s]+$)/i)[1].replace(cutOff, '');
+  const name = getParamName(from);
+  const path = parseQueryPartial(from, name);
```
Comment (uapasha) [inline lib/sql_parser.js]: @pj-infest please move definition of functions before usage, or use `function declaration`

Comment (uapasha) [inline lib/sql_parser.js]: @pj-infest oh, ok, that was done

**Pull request merged**